amgut/lib/data_access/ag_data_access.py

```diff
@@ -323,8 +340,11 @@ def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
                 self.deleteSample(info['barcode'], ag_login_id)
 
             # Delete last due to foreign keys
-            sql = "DELETE FROM ag_kit_barcodes WHERE survey_id IN %s"
+            sql = """DELETE FROM ag.source_barcodes_surveys
+                     WHERE survey_id IN %s"""
             TRN.add(sql, [tuple(survey_ids)])
+            sql = "DELETE FROM ag.ag_kit_barcodes WHERE barcode IN %s"
```

Comment:
when we originally put this method together, we did not have multiple surveys in this fashion, so i think this may be a case of an outdated comment :/ i think the most correct thing is to a) remove the specific relation in `source_barcodes_surveys` and b) if and only if the associated barcodes are no longer associated to any survey in `source_barcodes_surveys` to then remove the barcode entry from `ag_kit_barcodes`. Does that make sense?